File: backend/app/schemas/interview.py
from pydantic import BaseModel
from typing import Optional, List, Any
from datetime import datetime


class InterviewCreate(BaseModel):
    interview_id: str
    job_position: str
    job_description: str
    company_name: str = ""
    company_details: str = ""
    duration: str = "10"
    type: str = ""
    question_list: Any = None
    company_summary: str = ""


class InterviewUpdate(BaseModel):
    job_position: Optional[str] = None
    job_description: Optional[str] = None
    company_name: Optional[str] = None
    company_details: Optional[str] = None
    duration: Optional[str] = None
    type: Optional[str] = None
    question_list: Optional[Any] = None
    company_summary: Optional[str] = None


class InterviewResponse(BaseModel):
    id: int
    interview_id: str
    user_email: str
    job_position: Optional[str] = None
    job_description: Optional[str] = None
    company_name: Optional[str] = None
    company_details: Optional[str] = None
    duration: Optional[str] = None
    type: Optional[str] = None
    question_list: Optional[Any] = None
    company_summary: Optional[str] = None
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class QuestionGenerateRequest(BaseModel):
    jobPosition: str
    jobDescription: str
    duration: str
    type: str
    companyDetails: str = ""


class FeedbackGenerateRequest(BaseModel):
    conversation: str
    interview_id: str
    user_email: str
    call_id: str = ""
    companyDetails: str = ""


class CompanySummaryRequest(BaseModel):
    jobPosition: str
    jobDescription: str
    companyDetails: str


class FeedbackCreate(BaseModel):
    interview_id: str
    user_email: str
    user_name: str = ""
    feedback: Any = None
    transcript: Any = None
    call_id: str = ""


class FeedbackResponse(BaseModel):
    id: int
    interview_id: str
    user_email: str
    user_name: Optional[str] = None
    feedback: Optional[Any] = None
    transcript: Optional[Any] = None
    call_id: Optional[str] = None
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class SaveTranscriptRequest(BaseModel):
    callId: str
    interviewId: str
    userEmail: str
    userName: str = ""

File: frontend/tests/fixtures/interview.ts
export const MOCK_INTERVIEW_ID = 'mock-interview-abc123'

export const MOCK_INTERVIEW = {
  id: 1,
  interview_id: MOCK_INTERVIEW_ID,
  job_position: 'Senior Frontend Developer',
  company_name: 'TechCorp',
  duration: '30',
  type: '["Technical", "Behavioral"]',
  question_list: [
    { question: 'Tell me about your React experience', type: 'Technical' },
    { question: 'Describe a challenging project you worked on', type: 'Behavioral' },
  ],
  company_details: 'TechCorp builds innovative web products.',
  company_summary: 'TechCorp is an innovative technology company focused on building great products.',
  created_at: '2024-01-15T10:00:00Z',
}

export const MOCK_QUESTIONS_RESPONSE = {
  content: JSON.stringify({
    interviewQuestions: [
      { question: 'Tell me about your React experience', type: 'Technical' },
      { question: 'Describe a challenging project you worked on', type: 'Behavioral' },
      { question: 'How do you handle state management in large applications?', type: 'Technical' },
    ],
  }),
}

export const MOCK_FEEDBACK_LIST = [
  {
    id: 1,
    interview_id: MOCK_INTERVIEW_ID,
    user_email: 'candidate@example.com',
    user_name: 'Jane Doe',
    feedback: {
      feedback: {
        rating: {
          technicalSkills: 8,
          communicationSkills: 7,
          problemSolving: 8,
          experience: 7,
          OverallRating: 8,
        },
        summary: ['Strong technical skills', 'Good communication'],
        Recommendation: 'recommended',
        RecommendationMessage: 'Strong candidate with solid React skills',
      },
    },
    transcript: '[]',
    call_id: 'call-test-id',
    created_at: '2024-01-16T10:00:00Z',
  },
]

export const MOCK_FEEDBACK_AI_RESPONSE = {
  content: JSON.stringify({
    feedback: {
      rating: {
        technicalSkills: 8,
        communicationSkills: 7,
        problemSolving: 8,
        experience: 7,
        OverallRating: 8,
      },
      summary: ['Strong technical skills', 'Good communication'],
      Recommendation: 'recommended',
      RecommendationMessage: 'Strong candidate',
    },
  }),
}
